scripts/vm/network/security_group.py

```diff
@@ -36,6 +39,20 @@
 hyper = cfo.getEntry("hypervisor.type")
 if hyper == "lxc":
     driver = "lxc:///"
+
+lock_handle = None
+
+def obtain_file_lock(path):
+    global lock_handle
+    lock_handle = open(path, 'w')
+    try:
+        fcntl.lockf(lock_handle, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
- you may consider a `with` open() syntax that is more pythonic
- who's closing/expiring these file locks, after the work is done the lock needs to be freed/closed